### ISSUE TITLE
java.lang.UnsupportedOperationException: Not implemented.

### ISSUE THREAD

**Issue by showaid** [Erudika/scoold#37]
Hello @albogdano,

I hope you have a good day.

Today, I got an exception when I tried to remove an user forcefully.
It's UnsupportedOperationException:
```
...
Caused by: java.lang.UnsupportedOperationException: Not implemented.
	at com.erudika.para.search.MockSearch.findTerms(MockSearch.java:190)
	at com.erudika.para.core.User.getIdentifiers(User.java:365)
	at com.erudika.para.core.User.delete(User.java:354)
        at com.erudika.scoold.controllers.AdminController.forceDelete(AdminController.java:127)

...
```
I don't exactly know why, but scoold tries to use MockSearch. It should've been LuceneSearch, right?

The para system generated a log message that it uses LuceneSearch instance.

`2018-03-07 11:56:47 [34m[INFO ][0;39m Loaded new DAO, Search and Cache implementations - AWSDynamoDAO, LuceneSearch and CaffeineCache.
`

Could you take a look at this?

Thank you!

FYI. The scoold version I tested is the most recent commit.

**Comment by showaid** [comment]
I found out that para client is required to delete sobject instead of using it directly(sobject.delete()).

`
public String forceDelete(@RequestParam Boolean confirmdelete, @RequestParam String id, HttpServletRequest req) {
	Profile authUser = utils.getAuthUser(req);
	if (confirmdelete && utils.isAdmin(authUser)) {
		ParaObject sobject = pc.read(id);
		if (sobject != null) {
			sobject.delete();
			logger.info("{} #{} deleted {} #{}", authUser.getName(), authUser.getId(),
						sobject.getClass().getName(), sobject.getId());
		}
	}
	return "redirect:" + ADMINLINK;
}
`
I changed the code from sobject.delete() to pc.delete(sobject). And the user profile shows that it is deleted user after the deletion. 

![screenshot - 2018-03-07 2_11_05](https://user-images.githubusercontent.com/1582582/37074683-62713ba6-2211-11e8-8a5f-842d3a916e00.png)

However, the user can sign in again and it creates another user account which is using the same id.

![screenshot - 2018-03-07 2_16_26](https://user-images.githubusercontent.com/1582582/37074931-955f29dc-2212-11e8-95a7-04548ff1639d.png)

My environment is using an AD server to authenticate. And maybe that's the reason why.

Thank you!
